OpenRA.Game/Network/OrderManager.cs

```diff
@@ -21,9 +21,23 @@ namespace OpenRA.Network
 {
 	public sealed class OrderManager : IDisposable
 	{
+		const double CatchupFactor = 0.1;
+		const double CatchUpLimit = 1.5;
+		const int SimLagThreshold = 500;
+		const int SimLagLimit = 1000;
+
 		readonly SyncReport syncReport;
 
-		readonly Dictionary<int, Queue<byte[]>> pendingPackets = new Dictionary<int, Queue<byte[]>>();
+		readonly Dictionary<int, Queue<(byte[], int)>> pendingPackets = new Dictionary<int, Queue<(byte[], int)>>();
```

Comment:
We probably want to name the tuple in the Queue instead of using Item1 and Item2.